proto/viam/service/mlmodel/v1/mlmodel.proto

```diff
@@ -91,3 +98,70 @@ enum LabelType {
   // the position of the tensor value in the axis is the label index
   LABEL_TYPE_TENSOR_AXIS = 2;
 }
+
+message FlatTensorDataInt8 {
+  bytes data = 1;
+}
+
+message FlatTensorDataUInt8 {
+  bytes data = 1;
+}
+
+message FlatTensorDataInt16 {
+  // Pack two 16-bit numbers per entry.
+  // TODO: Describe handling for BE
+  repeated fixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataUInt16 {
+  // Pack two 16-bit numbers per entry.
+  // TODO: Describe handling for BE
+  repeated fixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataInt32 {
+  repeated sfixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataUInt32 {
+  repeated fixed32 data = 1 [packed = true];
+}
+
+message FlatTensorDataInt64 {
+  repeated sfixed64 data = 1 [packed = true];
+}
+
+message FlatTensorDataUInt64 {
+  repeated fixed64 data = 1 [packed = true];
+}
+
+message FlatTensorDataFloat {
+  repeated float data = 1 [packed = true];
+}
+
+message FlatTensorDataDouble {
+  repeated double data = 1 [packed = true];
+}
+
+message FlatTensor {
+  // TODO
+  repeated fixed64 shape = 1;
+
+  // TODO
+  oneof tensor {
+    FlatTensorDataInt8 int8_tensor = 2;
```

Comment:
I think the idea of doing a more general FlatTensorData over bytes is nice but I'm thinking about the decoding.  We could use a simple string (bytes) as the type code (it'd read "uint32" or "float16"). But are we sure that there's a standard way to unpack each of these primitive types from bytes? If there's a single "common sense" way that'll make sense for any SDK then we should really consider it. But I don't think there is. And we definitely shouldn't have a secret decoder.

Reply:
@kharijarrett - 

The idea I was after was that if the inference engine had its own space of types, like this https://github.com/tensorflow/tensorflow/blob/bb6fb6d2b7c56da81d0f183788a963243d28b448/tensorflow/core/framework/types.proto#L13.

Then let's say you wanted to send a buffer of complex numbers over to the tensorflow side. We don't have direct support for that in our proto defs, nor are we ever likely to support every type that every inference engine can handle.

But if we had the "back door" I'm proposing, the client could just send along a buffer of bytes which it had "correctly prepared" to contain bytes that the tensorflow side could use as complex, and then also send along the "foreign type code" 8, which is what TensorFlow uses for `DT_COMPEX`. Basically, a client which had special knowledge of the backend in use could tunnel typed data through us.

Reply:
kServe has a repeated bytes for tensor content as an option, as well as a lot of different "escape hatches" for various representations of inputs and output tensors. But given that it can be added in later and be backwards compatible, it's not necessary to include it now.

Reply:
Agreed; we can investigate further changes along those lines later.